packages/@aws-cdk/aws-ec2/lib/peer.ts

```diff
@@ -75,6 +75,13 @@ export class Peer {
     return new PrefixList(prefixListId);
   }
 
+  /**
+   * A security group ID
+   */
+  public static securityGroupId(securityGroupId: string, sourceSecurityGroupOwnerId?: string): IPeer {
```

Comment:
I think there's an argument that the `sourceSecurityGroupOwnerId` parameter should be mandatory. If the parameter is omitted, this generates an identical rule config to a SecurityGroup itself, which doesn't really provide (much) value.

Do you see a compelling reason to keep it optional?

Reply:
The only reason I can think to keep `sourceSecurityGroupOwnerId` optional is to mimic the behavior of the `AWS::EC2::SecurityGroup` resource type in [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group-rule-1.html#cfn-ec2-security-group-rule-sourcesecuritygroupownerid). Users are not required to put a `SourceSecurityGroupOwnerId` if the security group is in the same account. I agree that this duplicates existing functionality in CDK, but it does provide additional flexibility for how a rule containing a security group id can be added to a new security group in CDK. 

Requiring the `sourceSecurityGroupOwnerId` parameter will not cause any issues if a user specifies a security group which exists in the same account as the new `SecurityGroup`. Requiring the parameter will just slightly deviate from the behavior outlined in CloudFormation.

Reply:
@njlynch What do you think of the above reasoning?

Reply:
Fair enough. It's a bit duplicative, but not harming anything. Thanks.